cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/BuiltInRolesAdministrationTest.scala

```diff
@@ -0,0 +1,329 @@
+          resultTable()
+        }
+      }
+      section("How to recreate the `PUBLIC` role", "administration-built-in-roles-public-recreate", "enterprise-edition") {
+        initQueries(
+          """REVOKE GRANT ACCESS ON DEFAULT DATABASE FROM PUBLIC""")  // setup so that later when the grant query gets executed, it will show systemUpdates: 1
+        p(
+          """The `PUBLIC` role can not be dropped and thus there is no need to recreate the role itself.
+            |To restore the role to its original capabilities two steps are needed: First, all `GRANT` or `DENY` on this role should be revoked (see output of `SHOW ROLE PUBLIC PRIVILEGES` on what to revoke).
+            |Secondly, the following query need to be executed:""".stripMargin)
+        query("GRANT ACCESS ON DEFAULT DATABASE TO PUBLIC", ResultAssertions((r) => {
+          assertStats(r, systemUpdates = 1)
+        })) {
+          statsOnlyResultTable()
+        }
+
+        p("The resulting `PUBLIC` role has the same privileges as the " +
+          "<<administration-built-in-roles-public-privileges,original built-in `PUBLIC` role>>.")
+
+//        query("SHOW ROLE PUBLIC PRIVILEGES", ResultAssertions(p => true)) {
+//          resultTable()
+//        }
+      }
+    }
+    section("The `reader` role", "administration-built-in-roles-reader", "enterprise-edition") {
+      p("The `reader` can perform read-only queries on all databases except `system`.")
+      section("Privileges of the `reader` role", "administration-built-in-roles-reader-privileges", "enterprise-edition") {
+        query("SHOW ROLE reader PRIVILEGES", ResultAssertions(p => true)) {
+          resultTable()
+        }
+      }
+      section("How to recreate the `reader` role", "administration-built-in-roles-reader-recreate", "enterprise-edition") {
+        initQueries(
+          "DROP ROLE reader")  // setup so that later when the grant query gets executed, it will show systemUpdates: 1
+        p(
+          """
+            |To restore the role to its original capabilities two steps are needed: First, if not already done, execute `DROP ROLE reader`.
+            |Secondly, the following queries need to be executed:""".stripMargin)
+
+        query("CREATE ROLE reader", ResultAssertions((r) => {
+          assertStats(r, systemUpdates = 1)
+        })) {
+          statsOnlyResultTable()
+        }
+        query("GRANT ACCESS ON DATABASE * TO reader", ResultAssertions((r) => {
+          assertStats(r, systemUpdates = 1)
+        })) {
+          statsOnlyResultTable()
+        }
+        query("GRANT MATCH {*} ON GRAPH * NODES * TO reader", ResultAssertions((r) => {
+          assertStats(r, systemUpdates = 1)
+        })) {
+          statsOnlyResultTable()
+        }
+        query("GRANT MATCH {*} ON GRAPH * RELATIONSHIPS * TO reader", ResultAssertions((r) => {
+          assertStats(r, systemUpdates = 1)
+        })) {
+          statsOnlyResultTable()
+        }
```

Comment:
This can be done with just one command: `GRANT MATCH {*} ON GRAPH * ELEMENTS * TO reader` or just `GRANT MATCH {*} ON GRAPH * TO reader`. The planner will split it up to nodes and relationships internally so the privileges should be identical (just as it does for WRITE)

Also true for the other roles with match privileges.